lib/src/main/java/io/cloudquery/helper/ArrowHelper.java

```diff
@@ -13,22 +16,126 @@
+      return;
+    }
+    if (vector instanceof BitVector) {
+      ((BitVector) vector).set(0, (int) data);
+      return;
+    }
+    if (vector instanceof FixedSizeBinaryVector) {
+      ((FixedSizeBinaryVector) vector).set(0, (byte[]) data);
+      return;
+    }
+    if (vector instanceof Float4Vector) {
+      ((Float4Vector) vector).set(0, (float) data);
+      return;
+    }
+    if (vector instanceof Float8Vector) {
+      ((Float8Vector) vector).set(0, (double) data);
+      return;
+    }
+    if (vector instanceof IntVector) {
+      ((IntVector) vector).set(0, (int) data);
+      return;
+    }
+    if (vector instanceof LargeVarBinaryVector) {
+      ((LargeVarBinaryVector) vector).set(0, (byte[]) data);
+      return;
+    }
+    if (vector instanceof LargeVarCharVector) {
+      ((LargeVarCharVector) vector).set(0, (Text) data);
+      return;
+    }
+    if (vector instanceof SmallIntVector) {
+      ((SmallIntVector) vector).set(0, (short) data);
+      return;
+    }
+    if (vector instanceof TimeStampVector) {
+      ((TimeStampVector) vector).set(0, (long) data);
+      return;
+    }
+    if (vector instanceof TinyIntVector) {
+      ((TinyIntVector) vector).set(0, (byte) data);
+      return;
+    }
+    if (vector instanceof UInt1Vector) {
+      ((UInt1Vector) vector).set(0, (byte) data);
+      return;
+    }
+    if (vector instanceof UInt2Vector) {
+      ((UInt2Vector) vector).set(0, (short) data);
+      return;
+    }
+    if (vector instanceof UInt4Vector) {
+      ((UInt4Vector) vector).set(0, (int) data);
+      return;
+    }
+    if (vector instanceof UInt8Vector) {
+      ((UInt8Vector) vector).set(0, (long) data);
+      return;
+    }
+    if (vector instanceof VarBinaryVector) {
+      ((VarBinaryVector) vector).set(0, (byte[]) data);
+      return;
+    }
+    if (vector instanceof VarCharVector) {
+      ((VarCharVector) vector).set(0, (Text) data);
+      return;
+    }
+    if (vector instanceof UUIDVector) {
+      ((UUIDVector) vector).set(0, (java.util.UUID) data);
+      return;
+    }
+    if (vector instanceof JSONVector) {
+      ((JSONVector) vector).setSafe(0, (byte[]) data);
+      return;
+    }
+
+    throw new IllegalArgumentException("Unsupported vector type: " + vector.getClass());
+  }
+
   public static ByteString encode(Table table) throws IOException {
     try (BufferAllocator bufferAllocator = new RootAllocator()) {
       Schema schema = toArrowSchema(table);
-      VectorSchemaRoot schemaRoot = VectorSchemaRoot.create(schema, bufferAllocator);
-      try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-        try (ArrowStreamWriter writer =
-            new ArrowStreamWriter(schemaRoot, null, Channels.newChannel(out))) {
-          writer.start();
-          writer.end();
-          return ByteString.copyFrom(out.toByteArray());
+      try (VectorSchemaRoot schemaRoot = VectorSchemaRoot.create(schema, bufferAllocator)) {
```

Comment:
Apparently we need to close `VectorSchemaRoot` (which closes all the underlying vectors) to avoid memory leaks. It's only an issue when we set data on the vectors so we I didn't see it on the table encode before.